MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -83,34 +90,68 @@ double norm(PETScVector const& x)
 void setVector(PETScVector& v,
                std::initializer_list<double> values)
 {
-    (void) v; (void) values;
-    // TODO implement
-}
+    std::vector<double> const vals(values);
+    std::vector<PETScVector::IndexType> idcs(vals.size());
+    std::iota(idcs.begin(), idcs.end(), 0);
 
+    v.set(idcs, vals);
+}
 
 void setMatrix(PETScMatrix& m,
-               PETScMatrix::IndexType const rows,
-               PETScMatrix::IndexType const cols,
                std::initializer_list<double> values)
 {
-    (void) m; (void) rows; (void) cols; (void) values;
-    // TODO implement
+    m.setZero();
+    addToMatrix(m, values);
 }
 
 void setMatrix(PETScMatrix& m, Eigen::MatrixXd const& tmp)
 {
+    using IndexType = PETScMatrix::IndexType;
+
+    auto const rows = tmp.rows();
+    auto const cols = tmp.cols();
+
+    assert(rows == m.getNRows() && cols == m.getNCols());
+
+    m.setZero();
+    std::vector<IndexType> row_idcs(rows);
+    std::vector<IndexType> col_idcs(cols);
 
-    (void) m; (void) tmp;
-    // TODO implement
+    std::iota(row_idcs.begin(), row_idcs.end(), 0);
+    std::iota(col_idcs.begin(), col_idcs.end(), 0);
+
+    // PETSc wants row-major
+    Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> tmp_ = tmp;
+
+    m.add(row_idcs, col_idcs, tmp_);
 }
 
 void addToMatrix(PETScMatrix& m,
-                 PETScMatrix::IndexType const rows,
-                 PETScMatrix::IndexType const cols,
                  std::initializer_list<double> values)
 {
-    (void) m; (void) rows; (void) cols; (void) values;
-    // TODO implement
+    using IndexType = PETScMatrix::IndexType;
+
+    auto const rows = m.getNRows();
+    auto const cols = m.getNCols();
+
+    assert((IndexType) values.size() == rows*cols);
+
+    Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> tmp(rows, cols);
+
+    auto it = values.begin();
+    for (IndexType r=0; r<rows; ++r) {
```

Comment:
Sure, it would be. But this file is currently only used within very small test cases. So, performance is not an issue. When reworking the matrix/vector classes there will have to be a different solution for this anyway.